pkg/argocd/argocd.go

```diff
@@ -417,6 +421,18 @@ func GetImagesFromApplication(app *v1alpha1.Application) image.ContainerImageLis
 		images = append(images, image)
 	}
 
+	annotations := app.Annotations
+	if updateImage, ok := annotations[common.ImageUpdaterAnnotation]; ok {
```

Comment:
Could you please add some explaining commentary to this block? It's easy to understand in the context of this pull request, but I think it might be worth to add some inline documentation explaining why we're doing this so people without the context will understand later on.